usr/lib/common/new_host.c

```diff
@@ -3380,20 +3380,17 @@ CK_RV SC_DigestEncryptUpdate(STDLL_TokData_t *tokdata,
                              CK_ULONG ulPartLen, CK_BYTE_PTR pEncryptedPart,
                              CK_ULONG_PTR pulEncryptedPartLen)
 {
-    UNUSED(sSession);
-    UNUSED(pPart);
-    UNUSED(ulPartLen);
-    UNUSED(pEncryptedPart);
-    UNUSED(pulEncryptedPartLen);
+    CK_RV rc;
 
-    if (tokdata->initialized == FALSE) {
-        TRACE_ERROR("%s\n", ock_err(ERR_CRYPTOKI_NOT_INITIALIZED));
-        return CKR_CRYPTOKI_NOT_INITIALIZED;
-    }
+    rc = SC_EncryptUpdate(tokdata, sSession, pPart, ulPartLen,
+                          pEncryptedPart, pulEncryptedPartLen);
+    if (rc != CKR_OK || pEncryptedPart == NULL)
+        goto out;
```

Comment:
What if pEncryptedPart != NULL but ulPartLen == 0?

Reply:
Then C_DigestEncryptUpdate was called with a zero length update. That is a valid case. Adding zero bytes to a digest is allowed (but will of course not update the digest).